src/tests/unit/tests/common/components/body-class-modifier.test.tsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { shallow } from 'enzyme';
+import * as React from 'react';
+
+import { BodyClassModifier } from 'common/components/body-class-modifier';
+import { DocumentManipulator } from 'common/document-manipulator';
+
+describe('BodyClassModifier', () => {
+    let documentManipulator: DocumentManipulator;
+
+    beforeEach(() => {
+        documentManipulator = { bodyClassNames: [] } as DocumentManipulator;
+    });
+
+    it('should render as null', () => {
+        const testSubject = shallow(<BodyClassModifier classNames={['any']} documentManipulator={documentManipulator} />);
+        expect(testSubject.getElement()).toBeNull();
+    });
+
+    it('should not modify class names it is not responsible for', () => {
+        documentManipulator.bodyClassNames = ['pre-existing'];
+
+        const testSubject = shallow(<BodyClassModifier classNames={['from-test-subject']} documentManipulator={documentManipulator} />);
+
+        expect(documentManipulator.bodyClassNames).toContain('pre-existing');
+
+        testSubject.setProps({ classNames: [] });
+
+        expect(documentManipulator.bodyClassNames).toContain('pre-existing');
+    });
+
+    it('should add classNames to documentManipulator.bodyClassNames when mounted', () => {
+        shallow(<BodyClassModifier classNames={['test-class']} documentManipulator={documentManipulator} />).render();
+        expect(documentManipulator.bodyClassNames).toEqual(['test-class']);
+    });
+
+    it('should update classNames in documentManipulator.bodyClassNames when props are updated', () => {
+        const testSubject = shallow(<BodyClassModifier classNames={['from-original-props']} documentManipulator={documentManipulator} />);
+        testSubject.render();
```

Comment:
question: why do we need to explicitly call `render`? Isn't `shallow` "rendering" the component anyway?

Reply:
Good catch, this is leftover from when I was misunderstanding the lifecycle model and had implemented `componentDidUpdate` without `componentDidMount`. Removed.